starter-core/src/test/groovy/io/micronaut/starter/build/gradle/GradleBuildTestVerifier.groovy

```diff
@@ -86,6 +90,31 @@ class GradleBuildTestVerifier implements BuildTestVerifier {
         })
     }
 
+    @Override
+    boolean hasExclusion(String groupId, String artifactId, String excludedGroupId, String excludedArtifactId) {
+        hasExclusion(groupId, artifactId, excludedGroupId, excludedArtifactId, Scope.COMPILE)
+    }
+
+    @Override
+    boolean hasExclusion(String groupId, String artifactId, String excludedGroupId, String excludedArtifactId, Scope scope) {
+        Optional<String> gradleConfigurationNameOptional = GradleConfiguration.of(scope, language, testFramework, null).map { it.getConfigurationName() }
+        if (!gradleConfigurationNameOptional.isPresent()){
+            throw new ConfigurationException("cannot match " + scope + " to gradle configuration");
+        }
+        String gradleConfigurationName = gradleConfigurationNameOptional.get()
+        String pattern = /(?s).*${gradleConfigurationName}\("${groupId.replace(".", "\\.")}:${artifactId}"\)\s*\{(.+?)\}/
+        println pattern
+        def match = template =~ pattern
+        if (match.size() > 0) {
+            if (buildTool == BuildTool.GRADLE) {
```

Comment:
Rewrite: 

```groovy
if (buildTool == BuildTool.GRADLE) {
    return match[0][1].contains("exclude(group: \"${excludedGroupId}\", module: \"${excludedArtifactId}\")")
 } else {
     return match[0][1].contains("exclude(group = \"${excludedGroupId}\", module = \"${excludedArtifactId}\"")
}
```

as: 

```groovy
return match[0][1].contains(buildTool == BuildTool.GRADLE 
  ? "exclude(group: \"${excludedGroupId}\", module: \"${excludedArtifactId}\")"
  : "exclude(group = \"${excludedGroupId}\", module = \"${excludedArtifactId}\"")
```

Reply:
ping @sdelamo 

I did it a bit differently. See if that's OK